tools/bin/hawq

```diff
@@ -82,7 +83,13 @@ def main():
     if len(sys.argv) > 2:
         hawq_command = sys.argv[1]
         second_arg = sys.argv[2]
-        sub_args = " ".join(sys.argv[2:])
+        sub_args_list = sys.argv[2:]
+        # Password can have special characters like semicolon (;), quotes(", ') etc, convert input password to a string
+        if hawq_command == 'ssh-exkeys' and '-p' in sub_args_list:
+            password_index = sub_args_list.index('-p') + 1
+            if len(sub_args_list) > password_index:
+              sub_arg_list[password_index] = json.dumps(sub_args_list[password_index])
```

Comment:
a bit cautious about using json here. Tried this from python shell:
>>> json.dumps('foo zzz')
'"foo zzz"'
So it seems the call preserved single quotes but also added double quotes inside the value, which would essentially change the password, no ?